### PR TITLE
Updating tutorial samples to Dart v2

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

**Comment by domesticmouse** [comment]
PTAL @kwalrath @legalcodes

File: futures/async-await-catch-error/web/main.dart [hunk 1]
```diff
@@ -5,37 +5,42 @@
 import 'dart:html';
 import 'dart:async';
 
+DivElement output = querySelector('#output');
+PreElement pre = PreElement();
```
Comment (natebosch) [inline futures/async-await-catch-error/web/main.dart]: [nit] Don't need a type on the left.

File: anagram/pubspec.yaml [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
-dependencies:
-  browser: ^0.10.0
 
-homepage: https://www.dartlang.org/docs/tutorials/add-elements/
-gist: https://gist.github.com/Sfshaza/0532bfcb70bf5e4a900c
-dartpad: https://dartpad.dartlang.org/0532bfcb70bf5e4a900c
+environment:
+  sdk: '>=2.2.0 <3.0.0'
+
+
```
Comment (kevmoo) [inline anagram/pubspec.yaml]: extra line

File: futures/async-await-catch-error/web/main.dart [hunk 3]
```diff
@@ -5,37 +5,42 @@
...
 
 void printBaseballScore() {
-  print('Baseball score: Red Sox 10, Yankees 0');
+  pre.appendText('Baseball score: Red Sox 10, Yankees 0\n');
 }
 
 // Imagine that this function is more complex and slow. :)
 Future gatherNewsReports() async {
-  String path = 'https://www.dartlang.org/f/dailyNewsDigest.txt';
+  String path = 'https://dart.dev/f/dailyNewsDigest.txt';
```
Comment (natebosch) [inline futures/async-await-catch-error/web/main.dart]: consider omitting types on the left.

https://dart.dev/guides/language/effective-dart/design#avoid-type-annotating-initialized-local-variables

File: anagram/pubspec.yaml [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+dev_dependencies:
+  build_runner: ^1.1.2
+  build_web_compilers: ^1.0.0
+  pedantic: ^1.7.0
+
+homepage: https://dart.dev/tutorials/web/low-level-html/add-elements
```
Comment (kevmoo) [inline anagram/pubspec.yaml]: put this above dependencies

File: cmdline/pubspec.yaml [hunk 5]
```diff
@@ -2,21 +2,21 @@ name: cmdline
...
 dev_dependencies:
-  pedantic: ^1.0.0
+  pedantic: ^1.7.0
 
 executables:
   dcat:
   dgrep:
 
-homepage: https://www.dartlang.org/tutorials/server/cmdline
+homepage: https://dart.dev/tutorials/server/cmdline
```
Comment (kevmoo) [inline cmdline/pubspec.yaml]: Move towards top

File: streams/last_positive/pubspec.lock [hunk 6]
```diff
@@ -1,17 +1,348 @@
 # Generated by pub
-# See http://pub.dartlang.org/doc/glossary.html#lockfile
```
Comment (kevmoo) [inline streams/last_positive/pubspec.lock]: Weird? Seems to have been run on an old SDK

File: futures/sequential-processing/web/main.dart [hunk 7]
```diff
@@ -11,10 +12,10 @@ void main() {
       .then((cValue) => doSomethingWith(cValue));
 }
 
-Future expensiveA() => new Future.value('from expensiveA');
-Future expensiveB() => new Future.value('from expensiveB');
-Future expensiveC() => new Future.value('from expensiveC');
+Future expensiveA() => Future.value('from expensiveA');
```
Comment (natebosch) [inline futures/sequential-processing/web/main.dart]: How about `Future<String>` return type?

File: streams/last_positive/pubspec.lock (hunk 6 above)
Comment (kevmoo) [reply streams/last_positive/pubspec.lock]: Search for `# See https://www.dartlang.org`

File: futures/sequential-processing/web/main.dart [hunk 8]
```diff
@@ -3,6 +3,7 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:html';
 
 void main() {
```
Comment (natebosch) [inline futures/sequential-processing/web/main.dart]: Should we consider using `async` here?

https://dart.dev/guides/language/effective-dart/usage#prefer-asyncawait-over-using-raw-futures

File: httpserver/bin/basic_file_server.dart [hunk 9]
```diff
@@ -11,15 +11,17 @@
 import 'dart:async';
 import 'dart:io';
 import 'package:http_server/http_server.dart';
+import 'package:path/path.dart';
```
Comment (natebosch) [inline httpserver/bin/basic_file_server.dart]: Typically we import this `as p;`

File: streams/last_positive/pubspec.lock (hunk 6 above)
Comment (domesticmouse) [reply streams/last_positive/pubspec.lock]: I'm in the process of deleting and re-generating all of the `pubspec.lock` files but this line isn't changing.

```
$ pub --version
Pub 2.3.2
```

File: anagram/pubspec.yaml (hunk 2 above)
Comment (domesticmouse) [reply anagram/pubspec.yaml]: done

File: anagram/pubspec.yaml (hunk 4 above)
Comment (domesticmouse) [reply anagram/pubspec.yaml]: done

File: cmdline/pubspec.yaml (hunk 5 above)
Comment (domesticmouse) [reply cmdline/pubspec.yaml]: done

File: streams/last_positive/pubspec.lock (hunk 6 above)
Comment (kevmoo) [reply streams/last_positive/pubspec.lock]: Maybe it hasn't been synced yet. No worries...

File: anagram/pubspec.yaml [hunk 10]
```diff
@@ -1,8 +1,13 @@
...
-homepage: https://www.dartlang.org/docs/tutorials/add-elements/
-gist: https://gist.github.com/Sfshaza/0532bfcb70bf5e4a900c
-dartpad: https://dartpad.dartlang.org/0532bfcb70bf5e4a900c
+environment:
+  sdk: '>=2.2.0 <3.0.0'
+
+
+dev_dependencies:
+  build_runner: ^1.1.2
+  build_web_compilers: ^1.0.0
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply anagram/pubspec.yaml]: Yup, and as this was all based on Stagehand I also went and updated that as well.

https://github.com/dart-lang/stagehand/pull/614

File: futures/async-await-catch-error/web/main.dart (hunk 1 above)
Comment (domesticmouse) [reply futures/async-await-catch-error/web/main.dart]: done

File: futures/async-await-catch-error/web/main.dart (hunk 3 above)
Comment (domesticmouse) [reply futures/async-await-catch-error/web/main.dart]: done

File: futures/sequential-processing/web/main.dart (hunk 7 above)
Comment (domesticmouse) [reply futures/sequential-processing/web/main.dart]: done

File: futures/sequential-processing/web/main.dart (hunk 8 above)
Comment (domesticmouse) [reply futures/sequential-processing/web/main.dart]: It is cleaner to do with `async` and `await`, but this directory of samples seems to be all about using Futures API.

File: httpserver/bin/basic_file_server.dart (hunk 9 above)
Comment (domesticmouse) [reply httpserver/bin/basic_file_server.dart]: done

**Comment by domesticmouse** [comment]
@kevmoo do you want to merge this as is, or wait until @kwalrath gets back?

**Comment by kevmoo** [comment]
> @kevmoo do you want to merge this as is, or wait until @kwalrath gets back?

Probably best to wait until she gets back.

**Comment by domesticmouse** [comment]
PTAL @kwalrath, now that you are back =)

**Pull request merged**